frontend/ui/widget/keyvaluepage.lua

```diff
@@ -519,18 +526,34 @@ function KeyValuePage:init()
 
     self:_populateItems()
 
-    local content = OverlapGroup:new{
-        allow_mirroring = false,
-        dimen = self.inner_dimen:copy(),
-        VerticalGroup:new{
-            align = "left",
-            self.title_bar,
-            VerticalSpan:new{ width = span_height },
-            self.main_content,
-        },
-        page_return,
-        footer,
-    }
+    local content
+    if not self.single_page then
+        content = OverlapGroup:new{
+            allow_mirroring = false,
+            dimen = self.inner_dimen:copy(),
+            VerticalGroup:new{
+                align = "left",
+                self.title_bar,
+                VerticalSpan:new{ width = span_height },
+                self.main_content,
+            },
+            page_return,
+            footer,
+        }
+    else
+        content = OverlapGroup:new{
```

Comment:
No need for this if/else and big duplication - if I get it right that only footer is there or not.
As footer is last, no issue it being nil :)

And even if it wasn't last, you could have use:
```lua
previous_widget,
footer or VerticalSpan:new{width=0},
next_widget,
```